src/chat/Database.cpp

```diff
@@ -0,0 +1,38 @@
+// allkeys.cpp
+// chris busch
+
+#include "Database.hpp"
+
+namespace chat {
+
+Record *Database::add_record() {
+    Record *record;
+    if ((record = new Record) == nullptr) {
```

Comment:
per other comments, we try our best to not use raw pointers in modern C++. Again, probably a step on the way toward that :)

`new` cannot return `nullptr`. Might as well get rid of that, if nothing else :)